### PR TITLE
feat(ui-components) Add Checkbox component

### PR DESCRIPTION
**Pull request by robertpin**
Replace all checkboxes in the Settings Dialog
Convert some dialog tabs to TS

### CODE REVIEW

**Comment by saghul** [comment]
Is the AtlasKit checkbox used anywhere else now?

**Comment by robertpin** [comment]
> Is the AtlasKit checkbox used anywhere else now?

Not sure, but I plan on replacing all in future PRs

**Review by saghul — CHANGES_REQUESTED** [review]

File: react/features/base/ui/components/web/Checkbox.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                checked = { checked }
+                disabled = { disabled }
+                name = { name }
+                onChange = { onChange }
+                type = 'checkbox' />
+            <Icon
+                className = 'checkmark'
+                color = { disabled ? BaseTheme.palette.icon03 : BaseTheme.palette.icon01 }
+                size = { 18 }
+                src = { indeterminate ? IconHorizontalLine : IconCheckMark } />
```
[the remark this answers is not in the document]
Comment (robertpin) [reply react/features/base/ui/components/web/Checkbox.tsx:177]: Yes,  indeterminate is needed for Checkbox Group. I forgot to remove it

**Comment by saghul** [comment]
Can we drop `IconHorizontalLine` then?

**Review by saghul — APPROVED** [review]

**Pull request merged**